roles/provision_vcenter/tasks/main.yml

```diff
@@ -19,7 +19,7 @@
     msg: "{{ _vcenter_lookup.stderr | default(_vcenter_lookup.msg) }}"
   when: >-
     _vcenter_lookup.instance is not defined and
-    ('non-existing VM ' + provision_vcenter_vm_name) not in _vcenter_lookup.msg
+    _vcenter_lookup.msg is not match("Unable to gather information for non-existing VM .*")
```

Comment:
I personally don't like the way `vmware_guest_info` module return failure when the VM don't exists. I feel better would be to return empty instance info. Checking the ouput of the error message is hard to maintain. Can we for example use `vmware_rest.vcenter_vm_info` module to check if VM exists, I beleive they don't fail in that case, or at least I hope.

Reply:
Yea i agree with you. Ill check out the REST module. We could also add a ticket to update vmware_guest_info to always return the `instance` attribute?

Reply:
I tried out the rest module and it does not fail if no VM is found which is good. However, its kind of tricky. The module does not support a port parameter. It is also difficult to test, since in our integration tests the soap and rest APIs will have two different ports.

I can work around both but I think it will involve having different parameters for the soap or rest vcenter port, which seems confusing. Id like to hear your thoughts or ideas